modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -1363,6 +1363,152 @@ definition {
+
+		Role.openRolesAdmin();
+
+		LexiconEntry.gotoEntry(rowEntry = "Regular Role");
+
+		Navigator.gotoNavItem(navItem = "Define Permissions");
+
+		ObjectAdmin.searchInDefinePermissions(search = "Custom Objects 185");
+
+		Click(
+			key_wikiPageVersionNew = "action.ActionName",
+			locator1 = "WikiTable#SELECT_VERSION_NEW_VERSION_CHECKBOX");
+
+		Click(
+			key_wikiPageVersionNew = "Access in Control Panel",
+			locator1 = "WikiTable#SELECT_VERSION_NEW_VERSION_CHECKBOX");
+
+		Click(
+			key_wikiPageVersionNew = "View",
+			locator1 = "WikiTable#SELECT_VERSION_NEW_VERSION_CHECKBOX");
+
+		Click(
+			key_wikiPageVersionNew = "Permissions",
+			locator1 = "WikiTable#SELECT_VERSION_NEW_VERSION_CHECKBOX");
+
+		PortletEntry.save();
+
+		var objectId = JSONObject.getObjectId(objectName = "CustomObject185");
+
+		Permissions.definePermissionViaJSONAPI(
+			resourceAction = "VIEW_CONTROL_PANEL",
+			resourceName = 90,
+			roleTitle = "Regular Role",
+			roleType = "regular");
+
+		Permissions.definePermissionViaJSONAPI(
+			resourceAction = "ACCESS_IN_CONTROL_PANEL",
+			resourceName = "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_${objectId}",
+			roleTitle = "Regular Role",
+			roleType = "regular");
+
+		Permissions.definePermissionViaJSONAPI(
+			resourceAction = "VIEW",
+			resourceName = "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_${objectId}",
+			roleTitle = "Regular Role",
+			roleType = "regular");
+
+		Permissions.definePermissionViaJSONAPI(
+			resourceAction = "VIEW",
+			resourceName = "com.liferay.object.model.ObjectDefinition#${objectId}",
+			roleTitle = "Regular Role",
+			roleType = "regular");
+
+		task ("When the user go to set permissions in object definition permissions") {
+			JSONRole.assignRoleToUser(
+				roleTitle = "Regular Role",
+				userEmailAddress = "userea@liferay.com");
+
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject185");
+
+			Click(
+				key_entry = "Entry Test",
+				locator1 = "ObjectPortlet#ENTRY_KEBAB_MENU");
+
+			Click(
+				key_dropdowntItem = "Action Label",
+				locator1 = "ObjectPortlet#CHECK_UNCHECK_DROPDOWN_ITEM");
+
+			AssertTextEquals(
+				key_entryDetail = "New Entry Test",
+				key_settingHeader = "",
+				locator1 = "Account#DEFAULT_SETTINGS_TABLE_ITEM",
+				value1 = ${key_entryDetail});
+
+			task ("Then user should be able to manage standalone action permissions") {
+				Click(
+					key_newOption = "clay-dropdown-menu-6",
+					key_objectLabel = "Actions",
+					locator1 = "ObjectAction#OBJECT_KEBAB_OPTION");
+
+				Click(
+					key_menuItem = "Permissions",
+					locator1 = "MenuItem#ANY_MENU_ITEM");
+
+				if (IsElementPresent(locator1 = "Permissions#CONFIGURATION_PERMISSIONS_ROLE_NAME", value1 = "User")) {
+					Alert.viewSuccessMessage();
```

Comment:
The user should modify the permission of the action here and verify that the new permissions are in effect